chesster.js

```diff
@@ -644,6 +647,95 @@ controller.hears([
     });
 });
 
+controller.hears([
+    /pairing(?: @?([^\s]+))?/
+], [
+    'direct_mention',
+    'direct_message'
+], function(bot, message) {
+    bot_exception_handler(bot, message, function() {
+        // The user is either a string or an id
+        var nameOrId = message.match[1];
+        var requesting_player = users.getByNameOrID(message.user);
+
+        // The name or Id was provided, so parse it out
+        var player = null;
+        var player = users.getByNameOrID(nameOrId);
+
+        // If the player didn't exist that way, then it could be the @notation
+        if (!player) {
+            player = users.getByNameOrID(nameOrId.match(/<@([^\s]+)>/)[1]);
+        }
+
+        // If that still yields nothing, then must be the player from the message
+        if (!player) {
+            player = requesting_player;
+        }
+
+        preparePairingCompetitorMessage(requesting_player, player, function(response) {
+            bot.reply(message, response);
+        });
+    });
+});
+
+function uncaptain(playerName) {
+    return playerName.endsWith("*") ? playerName.substring(0, playerName.length - 1) : playerName;
+}
+
+function preparePairingCompetitorMessage(requesting_player, target_player, callback) {
```

Comment:
We have not been consistent in using camel or snake case. This is not your problem. It's something that I need to address. Just noting it here.